sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SyncPoller.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+import java.time.Duration;
+
+/**
+ * Type for doing synchronous polling of a long-running operation.
+ *
+ * @param <T> The type of poll response value.
+ * @param <U> The type of the final result of long-running operation.
+ */
+public interface SyncPoller<T, U> {
+    /**
+     * Poll once and return the poll response received.
+     *
+     * @return the poll response
+     */
+    PollResponse<T> poll();
+
+    /**
+     * Wait for polling to complete. The polling is considered complete based on status defined in
+     * {@link LongRunningOperationStatus}.
+     *
+     * @return the final poll response
+     */
+    PollResponse<T> waitForCompletion();
```

Comment:
In sync world, if this method is called, this will block indefinitely with no option for user to cancel. Should we always enforce a timeout for `waitUntil` methods?

Reply:
Discussed offline, we'll keep both methods `waitUntil()` and `waitUntil(Timeout)` to simplify the basic user experience. We also looked to see any other APIs has similar behavior - so far we identified the blocking calls in reactor [`Flux::blockFirst()`, `Flux::blockFirst(Timeout)`] and `Future.get()` `Future.get(Timeout)`.